### PR TITLE
feat(giveback): add causes breakdown block above the tabs

### PR DESCRIPTION
**Pull request by tsahimatsliah**
## What

Adds a **"Where the money will go"** breakdown block for the Giveback page that turns the community pool into a picture, sitting **below the hero and above the tabs**.

One component — `GivebackCausesBreakdown` — with a swappable `variant`:
- **stacked** — one pool split into colored segments on the hero funding-meter track (default)
- **donut** — SVG donut with the pool total in the hole
- **silos** — playful "storage tanks", each filled to its share of the top backer's level
- **mosaic** — treemap-lite tiles that flex-grow by share (balanced into two rows)

A `flat` prop drops the card chrome (border/surface/padding/glow) so it renders open in the page flow. All variants reuse the existing giveback food-palette accents and count-up motion.

The **flat donut** is wired into `GivebackPage`, gated on `showTabs && causesBreakdown.length > 0`, so it only appears for onboarded visitors with a funded pool.

## Reviewer notes

- ⚠️ **Placeholder split:** the contribution API has **no per-cause amounts** yet. `buildCausesBreakdown` spreads `currentCyclePoints` across the saved causes with a fixed descending weight purely to render the block — this should be replaced with a real backend per-cause field before it's truthful. The whole page stays behind the existing `featureGiveback` flag (default `isDevelopment`, off in prod) until then.
- No new feature flag — consistent with the rest of the giveback work, which is already gated by `featureGiveback`.
- `stroke="currentColor"` (driven by a `text-accent-*` class) is used for the donut arcs, since the `stroke-current` utility isn't generated in this Tailwind config.

## Testing

- Strict typecheck + ESLint clean.
- `GivebackCausesBreakdown.spec.tsx` covers shares/amounts, the optional description, all four variants, and empty/zero pools.
- Full giveback suite green (66 tests).
- Storybook page `Features/Giveback/Causes breakdown` covers every variant, the flat on-page layout, a gallery, and edge cases.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### Preview domain
https://claude-blissful-burnell-6e6f41.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/features/giveback/components/GivebackCausesBreakdown.tsx [hunk 1]
```diff
@@ -0,0 +1,477 @@
+import type { ReactElement, ReactNode } from 'react';
```
Comment (idoshamun) [inline packages/shared/src/features/giveback/components/GivebackCausesBreakdown.tsx:1]: You have multiple variations here. Remove everything that isn't picked

**Review by idoshamun — CHANGES_REQUESTED** [review]
You have conflicts with main. And also data is mocked so we can't merge until the API is completed

File: packages/shared/src/features/giveback/components/GivebackCausesBreakdown.tsx (hunk 1 above)
Comment (tsahimatsliah) [reply packages/shared/src/features/giveback/components/GivebackCausesBreakdown.tsx:1]: Done — removed the stacked-bar, storage-tanks and mosaic variants and the `variant` prop/switch. The component is now donut-only (the one that ships on the page). Also dropped the `GivebackBreakdownVariant` type, the `CauseEmblem` import, the mosaic-only accent tint, and trimmed the Storybook stories + spec to the donut.

**Comment by tsahimatsliah** [comment]
Thanks for the review — addressed both points:

- **Variants:** removed the stacked/tanks/mosaic variants and the `variant` prop; the component is donut-only now (see inline reply).
- **Conflicts with main:** merged latest `main` and resolved the `GivebackPage.tsx` conflict (kept your `overflow-x-clip` removal, re-applied the breakdown block on top).

On the **mocked data**: agreed it's a merge gate. The whole giveback route stays behind the `giveback` flag (defaults to `isDevelopment`, off in prod), and `buildCausesBreakdown` is a clearly-marked placeholder that splits the cycle pool by a fixed weight only to visualize the block — so no misleading data reaches users. Once the backend exposes a real per-cause amount, it's a one-function swap (that helper) with no component changes. Happy to hold the merge until the API lands, or gate the on-page render behind the real field if you'd prefer it dark until then.

Rechecking CI now.

**Review by idoshamun — APPROVED** [review]
